### ISSUE TITLE
no gps fix

### ISSUE THREAD

**Issue by BenCos17** [bitconnector/ttn_mapper_t-beam#24]
got this installed on my lilygo t beam (the 1.2 varient with the 2101 power chip) 
however I can't seem to get a gps fix 

```  
{
  "name": "as.up.data.forward",
  "time": "2024-03-18T13:01:26.872332427Z",
  "identifiers": [
    {
      "device_ids": {
        "device_id": "REDACTED",
        "application_ids": {
          "application_id": "REDACTED"
        },
        "dev_eui": "REDACTED",
        "join_eui": "REDACTED",
        "dev_addr": "REDACTED"
      }
    }
  ],
  "data": {
    "@type": "type.googleapis.com/ttn.lorawan.v3.ApplicationUp",
    "end_device_ids": {
      "device_id": "REDACTED",
      "application_ids": {
        "application_id": "REDACTED"
      },
      "dev_eui": "REDACTED",
      "join_eui": "REDACTED",
      "dev_addr": "REDACTED"
    },
    "correlation_ids": [
      "gs:uplink:REDACTED"
    ],
    "received_at": "2024-03-18T13:01:26.869019789Z",
    "uplink_message": {
      "session_key_id": "AY5RpvhBCFsBPJhNRm3Mdg==",
      "f_port": 1,
      "f_cnt": 1,
      "frm_payload": "QAIA",
      "decoded_payload": {
        "bat": 3.14,
        "gps": 0,
        "msg": "status: startup\nGPS: no fix\nbat: -34% 3.14V",
        "percentage": -34,
        "status": 2
      },
      "rx_metadata": [
        {
          "gateway_ids": {
            "gateway_id": "eui-afdcbacf1250e868",
            "eui": "AFDCBACF1250E868"
          },
          "timestamp": 3486013557,
          "rssi": -62,
          "channel_rssi": -62,
          "snr": 12.2,
          "frequency_offset": "3619",
          "uplink_token": REDACTED,
          "received_at": "2024-03-18T13:01:26.658027940Z"
        }
      ],
      "settings": {
        "data_rate": {
          "lora": {
            "bandwidth": 125000,
            "spreading_factor": 10,
            "coding_rate": "4/5"
          }
        },
        "frequency": "868100000",
        "timestamp": 3486013557
      },
      "received_at": "2024-03-18T13:01:26.665481628Z",
      "consumed_airtime": "0.329728s",
      "network_ids": {
        "net_id": "000013",
        "ns_id": "EC656E0000000181",
        "tenant_id": "ttn",
        "cluster_id": "eu1",
        "cluster_address": "eu1.cloud.thethings.network"
      }
    }
  },
  "correlation_ids": [
    "gs:uplink:REDACTED"
  ],
  "origin": "ip-10-100-12-78.eu-west-1.compute.internal",
  "context": {
    "tenant-id": "CgN0dG4="
  },
  "visibility": {
    "rights": [
      "RIGHT_APPLICATION_TRAFFIC_READ"
    ]
  },
  "unique_id": "REDACTED"```

**Comment by BenCos17** [comment]
"received_at": "2024-03-18T13:01:26.869019789Z",
    "uplink_message": {
      "session_key_id": "REDACTED==",
      "f_port": 1,
      "f_cnt": 1,
      "frm_payload": "QAIA",
      "decoded_payload": {
        "bat": 3.14,
        "gps": 0,
        "msg": "status: startup\nGPS: no fix\nbat: -34% 3.14V",
        "percentage": -34,
        "status": 2

**Comment by BenCos17** [comment]
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13192
load:0x40080400,len:3028
entry 0x400805e4
AXP2101 PMU init succeeded
3490
Wakeup caused by reset
Joining+ Package: 20 37 00 00 13 00 00 69 8d 0b 26 03 05 18 4f 84 e8 56 84 b8 5e 84 88 66 84 58 6e 84 00 10 b6 8f 66
.success!
+ Package: 40 69 8d 0b 26 00 01 00 01 1a 2b aa d3 c5 0a d4
          |hd|   addr    |Ct| FCnt|Pt|        |    MIC    |
ets Jul 29 2019 12:21:46

rst:0x5 (DEEPSLEEP_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13192
load:0x40080400,len:3028
entry 0x400805e4
AXP2101 PMU init succeeded
3513
Wakeup caused by timer
GPS-loop
GPS-end: no fix

**Comment by bitconnector** [comment]
Looks good to me except for your battery voltage. You should consider charging your device
For the GPS I would refer to this issue
<https://github.com/bitconnector/ttn_mapper_t-beam/issues/23#issuecomment-1817932574>

**Comment by BenCos17** [comment]
> Looks good to me except for your battery voltage. You should consider charging your device For the GPS I would refer to this issue <[#23 (comment)](https://github.com/bitconnector/ttn_mapper_t-beam/issues/23#issuecomment-1817932574)>

yep I have it on charge 
seems to have at least managed to have the gps led start blinking now which is a start at least
still no fix in ttn logs though for some reason 

```      "received_at": "2024-03-18T16:00:40.547665349Z",
    "uplink_message": {
      "session_key_id": "AY5SSw71TToeg2YiqWjo3A==",
      "f_port": 1,
      "f_cnt": 1,
      "frm_payload": "oAIA",
      "decoded_payload": {
        "bat": 4.1,
        "gps": 0,
        "msg": "status: startup\nGPS: no fix\nbat: 92% 4.1V",
        "percentage": 92,
        "status": 2
      }, ```

**Comment by BenCos17** [comment]
well replaced the antenna and now it seems to have a blinking gps led 
still shows no fix in the ttn console though for some reason

**Comment by BenCos17** [comment]
```mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13192
load:0x40080400,len:3028
entry 0x400805e4
AXP2101 PMU init succeeded
3421
Wakeup caused by timer
GPS-loop
GPS-end: no fix
ets Jul 29 2019 12:21:46

rst:0x5 (DEEPSLEEP_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13192
load:0x40080400,len:3028
entry 0x400805e4
AXP2101 PMU init succeeded
3436
Wakeup caused by timer
GPS-loop
GPS-end: no fix ```




``` "uplink_message": {
      "session_key_id": "REDACTED",
      "f_port": 1,
      "f_cnt": 1,
      "frm_payload": "UwIA",
      "decoded_payload": {
        "bat": 3.33,
        "gps": 0,
        "msg": "status: startup\nGPS: no fix\nbat: -9% 3.33V",
        "percentage": -9,
        "status": 2
      },
      "rx_metadata": [
        {
          "gateway_ids": {
            "gateway_id": "REDACTED",
            "eui": "REDACTED"
          },
          "timestamp": 2087609478,
          "rssi": -52,
          "channel_rssi": -52,
          "snr": 13.2,
          "frequency_offset": "4015",
          "uplink_token": "REDACTED",
          "received_at": "2024-03-26T11:35:15.209398360Z"
        }
      ], ```

**Comment by BenCos17** [comment]
and now it seems to be working so I'll close this issue 
I think it was a bad antenna on the board
